packages/web-ui/src/BaseCheckboxGroup/BaseCheckboxGroup.tsx

```diff
@@ -0,0 +1,136 @@
+import { Fieldset } from '../Fieldset';
+import { FieldsetLegend } from '../FieldsetLegend';
+import { HelperText } from '../HelperText';
+import { Flex } from '../Flex';
+import { mergeIds } from '../utils';
+import { useIds } from '../hooks';
+import { BaseCheckboxGroupProps } from './BaseCheckboxGroup.props';
+import { CheckboxGroupProps } from '../CheckboxGroup';
+import { BaseCheckboxGroupProvider } from './BaseCheckboxGroup.context';
+
+const componentName = 'BaseCheckboxGroup';
+
+const BaseCheckboxGroup = React.forwardRef<HTMLFieldSetElement, BaseCheckboxGroupProps>(
+  (
+    {
+      name,
+      defaultValue,
+      value: valueProp,
+      required = false,
+      disabled = false,
+      onValueChange,
+      id: providedId,
+      label,
+      helperText,
+      helperTextPosition = 'top',
+      showHelperTextIcon,
+      error,
+      errorMessage,
+      showErrorMessageIcon,
+      'aria-labelledby': ariaLabelledby,
+      'aria-describedby': ariaDescribedby,
+      'aria-errormessage': ariaErrorMessage,
+      children,
+      ...props
+    },
+    ref
+  ) => {
+    const { id, labelId, helperTextId, errorMessageId } = useIds({
+      providedId,
+      componentPrefix: 'checkboxgroup',
+    });
+    const showErrorMessage = Boolean(error && errorMessage);
+    const showTopHelperText = helperText && helperTextPosition === 'top';
+    const showBottomHelperText = helperText && helperTextPosition === 'bottom';
+
+    // useControllableState will handle whether controlled or uncontrolled
+    const [value = [], setValue] = useControllableState({
+      prop: valueProp,
+      defaultProp: defaultValue,
+      onChange: onValueChange,
+    });
+
+    const handleItemCheck = React.useCallback(
+      (itemValue: string) => setValue((prevValue = []) => [...prevValue, itemValue]),
+      [setValue]
+    );
+
+    const handleItemUncheck = React.useCallback(
+      (itemValue: string) =>
+        setValue((prevValue = []) => prevValue.filter(value => value !== itemValue)),
+      [setValue]
+    );
+
+    const ariaDescribedbyValue = mergeIds(
+      ariaDescribedby || !!helperText ? helperTextId : undefined,
+      ariaErrorMessage || showErrorMessage ? errorMessageId : undefined
+    );
+    const providerValue = {
+      name,
+      required,
+      disabled,
+      value,
+      onItemCheck: handleItemCheck,
+      onItemUncheck: handleItemUncheck,
+      hasGroupHelperText: !!helperText,
+      'aria-describedby': ariaDescribedbyValue,
+    };
+
+    return (
+      <Fieldset
+        ref={ref}
+        {...props}
+        disabled={disabled}
+        id={id}
+        data-disabled={disabled ? '' : undefined}
+        aria-errormessage={ariaErrorMessage || showErrorMessage ? errorMessageId : undefined}
+        aria-labelledby={ariaLabelledby || !!label ? labelId : undefined}
+        aria-invalid={showErrorMessage}
+        aria-describedby={ariaDescribedbyValue}
+      >
+        {label || showTopHelperText ? (
```

Comment:
I'm not overly happy with this solution, I'm sure there's a better way than all the ternary statements, but until I have more time to consider this then it will have to do.

Reply:
Would there be a scenario where you can have helper text top and bottom?

Reply:
No

Reply:
In chechbox/radio groups **NO** 🙅 
In Form Fields **YES** 🤓 

<img width="591" alt="Screenshot 2024-06-27 at 13 51 45" src="https://github.com/utilitywarehouse/design-systems/assets/1488514/f3f26beb-fa56-4117-8c85-e3eb36f48f63">

Reply:
FormFields will be a different component 👍🏻